clients/venice-admin-tool/src/test/java/com/linkedin/venice/TestAdminToolConsumption.java

```diff
@@ -37,28 +43,93 @@
 
 
 public class TestAdminToolConsumption {
+  private final PubSubTopicRepository pubSubTopicRepository = new PubSubTopicRepository();
+  private static final String SCHEMA_STRING = "\"string\"";
+  private static final String STORE_NAME = "test_store";
+
+  @Test
+  void testAdminToolAdminMessageConsumption() {
+    int assignedPartition = 0;
+    String topic = Version.composeRealTimeTopic(STORE_NAME);
+    PubSubTopicPartition pubSubTopicPartition =
+        new PubSubTopicPartitionImpl(pubSubTopicRepository.getTopic(topic), assignedPartition);
+    int adminMessageNum = 10;
+    int dumpedMessageNum = 2;
+    List<PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long>> pubSubMessageList =
+        prepareAdminPubSubMessageList(STORE_NAME, pubSubTopicPartition, adminMessageNum);
+    Map<PubSubTopicPartition, List<PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long>>> messagesMap = new HashMap<>();
+    messagesMap.put(pubSubTopicPartition, pubSubMessageList);
+    ApacheKafkaConsumerAdapter apacheKafkaConsumer = mock(ApacheKafkaConsumerAdapter.class);
+    when(apacheKafkaConsumer.poll(anyLong())).thenReturn(messagesMap, Collections.EMPTY_MAP);
+    List<DumpAdminMessages.AdminOperationInfo> adminOperationInfos =
+        DumpAdminMessages.dumpAdminMessages(apacheKafkaConsumer, "cluster1", 0, dumpedMessageNum);
+    Assert.assertEquals(adminOperationInfos.size(), dumpedMessageNum);
+  }
+
+  private List<PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long>> prepareAdminPubSubMessageList(
+      String storeName,
+      PubSubTopicPartition pubSubTopicPartition,
+      int messageNum) {
+    List<PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long>> pubSubMessageList = new ArrayList<>();
+    for (int i = 0; i < messageNum; i++) {
+      String keyString = "test";
+      byte[] serializedKey = TopicMessageFinder.serializeKey(keyString, SCHEMA_STRING);
+      KafkaKey kafkaKey = new KafkaKey(PUT, serializedKey);
+      KafkaMessageEnvelope messageEnvelope = new KafkaMessageEnvelope();
+      messageEnvelope.producerMetadata = new ProducerMetadata();
+      messageEnvelope.producerMetadata.messageTimestamp = 0;
+      messageEnvelope.producerMetadata.messageSequenceNumber = 0;
+      messageEnvelope.producerMetadata.segmentNumber = 0;
+      messageEnvelope.producerMetadata.producerGUID = new GUID();
+      Put put = new Put();
+      put.schemaId = AdminOperationSerializer.LATEST_SCHEMA_ID_FOR_ADMIN_OPERATION;
+      AdminOperationSerializer deserializer = new AdminOperationSerializer();
+      StoreCreation storeCreation = (StoreCreation) AdminMessageType.STORE_CREATION.getNewInstance();
+      storeCreation.clusterName = "clusterName";
+      storeCreation.storeName = storeName;
+      storeCreation.owner = "owner";
+      storeCreation.keySchema = new SchemaMeta();
+      storeCreation.keySchema.definition = SCHEMA_STRING;
+      storeCreation.keySchema.schemaType = SchemaType.AVRO_1_4.getValue();
```

Comment:
Just curious what's this schema type thing?

Reply:
This is just for mocking Admin Message that can be serialized and deserialized in this test case.